pkg/scale/varying_data_type_test.go

```diff
@@ -324,6 +326,48 @@ func Test_decodeState_decodeVaryingDataType(t *testing.T) {
 	}
 }
 
+func Test_encodeState_encodeCustomVaryingDataType(t *testing.T) {
+	for _, tt := range varyingDataTypeTests {
+		t.Run(tt.name, func(t *testing.T) {
+			es := &encodeState{fieldScaleIndicesCache: cache}
+			vdt := tt.in.(VaryingDataType)
+			cvdt := customVDT(vdt)
+			if err := es.marshal(cvdt); (err != nil) != tt.wantErr {
+				t.Errorf("encodeState.encodeStruct() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !reflect.DeepEqual(es.Buffer.Bytes(), tt.want) {
+				t.Errorf("encodeState.encodeStruct() = %v, want %v", es.Buffer.Bytes(), tt.want)
+			}
+		})
+	}
+}
+func Test_decodeState_decodeCustomVaryingDataType(t *testing.T) {
+	for _, tt := range varyingDataTypeTests {
+		t.Run(tt.name, func(t *testing.T) {
+			vdt, err := NewVaryingDataType(VDTValue{}, VDTValue1{}, VDTValue2{}, VDTValue3(0))
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
```

Comment:
Maybe use testify test libs? 🤔 
Or are you trying to keep without dependency so it's movable outside of this repo with zero-dependency?

```suggestion
			require.NoError(t, err)
```